Sources_v3/StreamChat/Workers/Background/AttachmentUploader.swift

```diff
@@ -57,7 +57,7 @@ class AttachmentUploader<ExtraData: ExtraDataTypes>: Worker {
             wasEmpty = pendingAttachmentIDs.isEmpty
             changes.pendingUploadAttachmentIDs.forEach { pendingAttachmentIDs.insert($0) }
         }
-        if wasEmpty {
+        if wasEmpty, !pendingAttachmentIDs.isEmpty {
```

Comment:
nit:
```
let notEmptyNow = !pendingAttachmentIDs.isEmpty
if wasEmpty && notEmptyNow {
```
I took me a while to understand why you don't have contradiction here (`wasEmpty = pendingAttachmentIDs.isEmpty`).
I think this small change will make intention more clear and readable

Reply:
Hm, it may be even better to just 
```
guard !changes.isEmpty else { return }
```
at beginning of function, so we would not lock context for no reason

Reply:
Fixed with `guard` solution